WordPress/Classes/ViewRelated/Blog/SiteManagement/DeleteSiteViewController.swift

```diff
@@ -0,0 +1,255 @@
+        super.viewDidLoad()
+
+        title = NSLocalizedString("Delete Site", comment: "Title of settings page for deleting a site")
+
+        tableView.cellLayoutMarginsFollowReadableWidth = true
+        tableView.estimatedSectionHeaderHeight = 100.0
+        tableView.sectionHeaderHeight = UITableViewAutomaticDimension
+
+        WPStyleGuide.configureColors(for: view, andTableView: tableView)
+    }
+
+    // MARK: - Actions
+
+    fileprivate func deleteSite() {
+        tableView.deselectSelectedRowWithAnimation(true)
+        present(confirmDeleteController(), animated: true, completion: nil)
+    }
+
+    fileprivate func contactSupport() {
+        tableView.deselectSelectedRowWithAnimation(true)
+
+        WPAppAnalytics.track(.siteSettingsStartOverContactSupportClicked, with: blog)
+        if HelpshiftUtils.isHelpshiftEnabled() {
+            setupHelpshift(blog.account!)
+
+            let metadata = helpshiftMetadata(blog)
+            HelpshiftSupport.showConversation(self, withOptions: metadata)
+        } else {
+            if let contact = URL(string: "https://support.wordpress.com/contact/") {
+                UIApplication.shared.open(contact)
+            }
+        }
+    }
+
+    // MARK: - Delete Site Helpers
+
+    /// Creates confirmation alert for Delete Site
+    ///
+    /// - Returns: UIAlertController
+    ///
+    fileprivate func confirmDeleteController() -> UIAlertController {
+        let confirmTitle = NSLocalizedString("Confirm Delete Site", comment: "Title of Delete Site confirmation alert")
+        let messageFormat = NSLocalizedString("Please type in \n\n%@\n\n in the field below to confirm. Your site will then be gone forever.", comment: "Message of Delete Site confirmation alert; substitution is site's host")
+        let message = String(format: messageFormat, blog.displayURL!)
+        let alertController = UIAlertController(title: confirmTitle, message: message, preferredStyle: .alert)
+
+        let cancelTitle = NSLocalizedString("Cancel", comment: "Alert dismissal title")
+        alertController.addCancelActionWithTitle(cancelTitle, handler: nil)
+
+        let deleteTitle = NSLocalizedString("Delete this site", comment: "Delete Site confirmation action title")
+        let deleteAction = UIAlertAction(title: deleteTitle, style: .destructive, handler: { action in
+            self.deleteSiteConfirmed()
+        })
+        deleteAction.isEnabled = false
+        alertController.addAction(deleteAction)
+
+        alertController.addTextField(configurationHandler: { textField in
+            textField.addTarget(self, action: #selector(DeleteSiteViewController.alertTextFieldDidChange(_:)), for: .editingChanged)
+        })
+
+        return alertController
+    }
+
+    /// Verifies site address as password for Delete Site
+    ///
+    func alertTextFieldDidChange(_ sender: UITextField) {
+        guard let deleteAction = (presentedViewController as? UIAlertController)?.actions.last else {
```

Comment:
We might also want to double check that the action's style is `destructive`, just to make doubly sure?